Sources/NIOPosix/BaseSocket.swift

```diff
@@ -84,10 +84,8 @@ extension sockaddr_storage {
     /// This will crash if `ss_family` != AF_INET!
     mutating func convert() -> sockaddr_in {
         precondition(self.ss_family == NIOBSDSocket.AddressFamily.inet.rawValue)
-        return withUnsafePointer(to: &self) {
-            $0.withMemoryRebound(to: sockaddr_in.self, capacity: 1) {
-                $0.pointee
-            }
+        return withUnsafeMutableBytes(of: &self) {
```

Comment:
I think this can be `withUnsafeBytes(of:)` (and elsewhere)

Reply:
This func is mutating because on older Swift releases we didn't have a version of `withUnsafeBytes` that did _not_ take an `inout` as an argument. It's possible we dropped all the Swift releases where that's true, let me check.

Reply:
Seems to be fine, I've updated the PR with removal of the mutatings.